Feature mission profile

GGiangreco opened the pull request:
This code provides the mission profile of a helicopter, calculating various flight parameters for an assigned mission such as fuel consumption, flight time, and performance metrics for different flight phases, including climb, cruise, descent, and hovering. Functions for different flight phases are imported from F2Rotor library.

GGiangreco committed:
Create Mission_profile

This code provides the mission profile of a helicopter, calculating various flight parameters for an assigned mission such as fuel consumption, flight time, and performance metrics for different flight phases, including climb, cruise, descent, and hovering. Functions for different flight phases are imported from F2Rotor library.

GGiangreco committed:
Create Mission_profile_testcase

Mission profile testcase example

GGiangreco committed:
Mission_profile_testcase.py

GGiangreco committed:
Mission_profile.py

EttoreSaetta commented on Mission_profile.py:
```diff
@@ -0,0 +1,260 @@
+
+# Graphics
+
+def create_table_figure(climb_data, hover_data, forward_data, Descend_data, forward_data2, fuel_data, gamma_deg, gamma_deg_d):
+    
+    headers = ["Segment", "Altitude \n(ft)", "Speed \n(knots - TAS)", "Distance \n(nm)", "Time \n(s)", 
+               "Power Rating \n(%)", "Fuel Flow \n(kg/h)", "Fuel Used \n(kg)", "Gross Weight \n(kg)"]
+    
+    rows = [
+        ["Warm-up", "-", "-", "-", "-", "-", "-", "-", "-"],
+        ["Climb", climb_data["Altitude"], climb_data["Speed "], climb_data["Distance"], climb_data["Time"], 
+         climb_data["Power Rating"], climb_data["Fuel Flow"], climb_data["Fuel Used"], climb_data["Gross Weight"]],
+        ["Cruise", forward_data["Altitude"], forward_data["Speed "], forward_data["Distance"], forward_data["Time"], 
+         forward_data["Power Rating"], forward_data["Fuel Flow"], forward_data["Fuel Used"], forward_data["Gross Weight"]],
+        ["Descend", Descend_data["Altitude"], Descend_data["Speed "], Descend_data["Distance"], Descend_data["Time"], 
+         Descend_data["Power Rating"], Descend_data["Fuel Flow"], Descend_data["Fuel Used"], Descend_data["Gross Weight"]],
+        ["Hover", hover_data["Altitude"], hover_data["Speed "], hover_data["Distance"], hover_data["Time"], 
+         hover_data["Power Rating"], hover_data["Fuel Flow"], hover_data["Fuel Used"], hover_data["Gross Weight"]],
+        ["Cruise", forward_data2["Altitude"], forward_data2["Speed "], forward_data2["Distance"], forward_data2["Time"], 
+         forward_data2["Power Rating"], forward_data2["Fuel Flow"], forward_data2["Fuel Used"], forward_data2["Gross Weight"]],
+        ["Final descent", "-", "-", "-", "-", "-", "-", "-", "-"]
+    ]
+    
+    fig, ax = plt.subplots(figsize=(12, 6)) 
+    ax.axis('tight')
```
Le funzioni di plotting devono andare nei test case

EttoreSaetta commented on TestCases/Mission_profile_testcase.py:
```diff
@@ -0,0 +1,128 @@
+Vinf_vec = np.linspace(0, 65, 10)                                                          # Asymptotic air speed [m/s].
+
+# Input for Helicopter class.
+class Helicopter:
+
+    Omega_r_mr = 25                                                                       # Main rotor angular velocity.
+    R_mr = 5.5                                                                                  # Disk rotor radius [m].
+    R_hub = 0.1 * R_mr                                                                           # Hub rotor radius [m].
+    c_mr = c_r + c_tw*(r_segn-r_segn[0])                                                                # Chord law [m].
+    N_mr = 4                                                                                         # Number of blades.
+    theta0 = 25                                                                             # Initial pitch angle [deg].
+    theta_end = 15                                                                                # Tip pitch angle [deg].
+    theta_tw = (theta_end-theta0)/(r_segn[-1]-r_segn[0])                                              # Pitch law [deg].
+    Cla = 2 * pi                                                                            # Lift slope curve [rad^-1].
+    Cd0 = 0.012                                                                            # Parasitic drag coefficient.
+
+    # Class attributes.
+    def __init__(self, Omega_r_mr, R_mr, R_hub, c_mr, N_mr, theta0, theta_tw, Cla, Cd0):
+        Helicopter.Omega_r_mr = Omega_r_mr
+        Helicopter.R_mr = R_mr
+        Helicopter.R_hub = R_hub
+        Helicopter.c_mr = c_mr
+        Helicopter.N_mr = N_mr
+        Helicopter.theta0 = theta0
+        Helicopter.theta_tw = theta_tw
+        Helicopter.Cla = Cla
+        Helicopter.Cd0 = Cd0
+
```
Esiste già una classe elicottero nella libreria.

EttoreSaetta commented on Mission_profile.py:
```diff
@@ -0,0 +1,260 @@
+                     V_infty, dist_f, Vinf_vec, N, r_segn, c, theta, W, height_f, R, A, f_A, Clalpha, Cd_mean,\
+                          Omega, N_rotors, F_Prandtl, height_d, height_f2, dist_f2):
+
+    atmosphere = Atmosphere(height_h)                                                                     # Initialize atmospheric properties at the hover altitude
```
I commenti sono fuori margini

EttoreSaetta commented on Mission_profile.py:
```diff
@@ -0,0 +1,260 @@
+    # Output vectors
+ 
+    climb_data = {
+            "Altitude": "To " + f"{round(height_c * 3.28)}",
+            "Speed ": round(TAS_c,1),
+            "Distance": round(dist_c * 0.00054,1),
+            "Time": round(time_c),
+            "Power Rating": "MCP",
+            "Fuel Flow": "-",
+            "Fuel Used": round(F_used_c,1),
+            "Gross Weight": round(GW_c),
```
Come mai i `round()`?
Cosa sono quei valori "hard-coded"? Sono Conversioni di unità di misura? se si specificare!

EttoreSaetta commented on Mission_profile.py:
```diff
@@ -0,0 +1,260 @@
+def Mission_profile(Helicopter, SFC, Pd, height_c, MTOW, Voo, Voo_d, height_h, time_h,\
+                     V_infty, dist_f, Vinf_vec, N, r_segn, c, theta, W, height_f, R, A, f_A, Clalpha, Cd_mean,\
+                          Omega, N_rotors, F_Prandtl, height_d, height_f2, dist_f2):
+
```
Manca il commento alla funzione con I/O

GGiangreco committed:
Update Mission_profile.py

Resolved issues regarding comments, plot and hard-coded values for measurement conversions

GGiangreco committed:
Update Mission_profile_testcase.py

Resolved issues on helicopter class function

GGiangreco committed:
Update Mission_profile.py

Resolved issues on comment margin

GGiangreco committed:
Update Mission_profile.py

Resolved issues on comment margin

GGiangreco committed:
Update Mission_profile.py

GGiangreco committed:
Update Mission_profile.py

Removed Round() for data vectors

GGiangreco committed:
Update Mission_profile_testcase.py

GGiangreco committed:
Update Mission_profile.py

GGiangreco committed:
Update Mission_profile_testcase.py
